src/Dfe.PlanTech.Web/Middleware/SecurityHeadersMiddleware.cs

```diff
@@ -0,0 +1,44 @@
+namespace Dfe.PlanTech.Web.Middleware;
+
+/// <summary>
+/// Middleware that sets various response headers for security reasons.
+/// </summary>
+public class SecurityHeadersMiddleware
+{
+  private readonly ILogger<SecurityHeadersMiddleware> _logger;
+  private readonly RequestDelegate _next;
+
+  public SecurityHeadersMiddleware(ILogger<SecurityHeadersMiddleware> logger, RequestDelegate next)
+  {
+    _logger = logger;
+    _next = next;
+  }
+
+  public async Task InvokeAsync(HttpContext context)
+  {
+    AddFramejackingPreventHeaders(context);
+
+    await _next(context);
+  }
+
+  /// <summary>
+  /// Adds headers that prevent framejacking vulnerability
+  /// </summary>
+  /// <param name="context"></param>
+  private static void AddFramejackingPreventHeaders(HttpContext context)
+  {
+    context.Response.Headers["X-Frame-Options"] = "Deny";
```

Comment:
happy for deny or would SAMEORIGIN do?

Reply:
Same origin would be sufficient, however we're not currently using IFrames for anything (nor am I aware of plans to do so), so IMO deny is best in this instance. Happy to change it if you think same origin is preferable?

Reply:
yeah deny is fine, I can't imagine we'll use frame or iframe